controller/authorize_blackbox_test.go

```diff
@@ -0,0 +1,198 @@
+package controller_test
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"testing"
+
+	"github.com/fabric8-services/fabric8-auth/app"
+	"github.com/fabric8-services/fabric8-auth/app/test"
+	. "github.com/fabric8-services/fabric8-auth/controller"
+	"github.com/fabric8-services/fabric8-auth/gormtestsupport"
+	testsupport "github.com/fabric8-services/fabric8-auth/test"
+	uuid "github.com/satori/go.uuid"
+
+	"github.com/goadesign/goa"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+type TestAuthorizeREST struct {
+	gormtestsupport.DBTestSuite
+}
+
+func TestRunAuthorizeREST(t *testing.T) {
+	suite.Run(t, &TestAuthorizeREST{DBTestSuite: gormtestsupport.NewDBTestSuite()})
+}
+
+func (rest *TestAuthorizeREST) UnSecuredController() (*goa.Service, *AuthorizeController) {
+	svc := testsupport.ServiceAsUser("Login-Service", testsupport.TestIdentity)
+	loginService := newTestKeycloakOAuthProvider(rest.Application)
+	return svc, &AuthorizeController{Controller: svc.NewController("AuthorizeController"), Auth: loginService, Configuration: rest.Configuration}
+}
+
+func (rest *TestAuthorizeREST) TestAuthorizeOK() {
+	t := rest.T()
+	svc, ctrl := rest.UnSecuredController()
+
+	redirect := "https://openshift.io"
+	clientID := "5dec5fdb-09e3-4453-b73f-5c828832b28e"
+	responseType := "code"
+	state := uuid.NewV4()
+
+	test.AuthorizeAuthorizeTemporaryRedirect(t, svc.Context, svc, ctrl, nil, clientID, redirect, responseType, nil, state)
+}
+
+func (rest *TestAuthorizeREST) TestAuthorizeCallbackOK() {
+	t := rest.T()
+	_, ctrl := rest.UnSecuredController()
+
+	rw := httptest.NewRecorder()
+	u := &url.URL{
+		Path: fmt.Sprintf("/api/authorize"),
```

Comment:
I think you need to split your test into two tests, one for `/api/authorize` and another for `/api/authorize/callback` .

Reply:
Both controllers are in the same file. Doesn't it make sense to keep the tests also in the same file?

Reply:
yes same file is fine. But have different test methods.

Reply:
I am not sure what you want. I think tests are already split. Lets discuss this.

Reply:
You should have 2 tests:

`TestAuthorizeOK()`
and
`TestAuthorizeCallbackOK()`

Reply:
So effectively you needn't split since you are using the response `/api/authorize` in `/api/authorize/callback` .
would be nice to have an explicit test written for `/api/authorize` even if it duplicates the first half of `TestAuthorizeCallbackOK`